localstripe/resources.py

```diff
@@ -2019,6 +2019,110 @@ def statement_descriptor(self):  # Support Stripe API <= 2018-02-05
         return Product._api_retrieve(self.product).statement_descriptor
 
 
+class Payout(StripeObject):
+    object = 'payout'
+    _id_prefix = 'po_'
+
+    def __init__(self, amount=None, currency=None, description=None,
+                 metadata=None, statement_descriptor=None, destination=None,
+                 method=None, source_type=None, status=None, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        amount = try_convert_to_int(amount)
+
+        try:
+            assert type(amount) is int and amount > 0
+            assert type(currency) is str and currency in ('eur')
+            if description is not None:
+                assert type(description) is str
+            if metadata is not None:
+                assert type(metadata) is dict
+            if statement_descriptor is not None:
+                assert type(statement_descriptor) is str \
+                    and len(statement_descriptor) <= 22
+            if method is not None:
+                assert type(method) is str \
+                    and method in ('standard', 'instant')
+            if source_type is not None:
+                assert type(source_type) is str
+            if status is not None:
+                assert type(status) is str \
+                    and status in ('paid', 'pending', 'failed')
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        # All exceptions must be raised before this point.
+        super().__init__()
+
+        self.amount = amount
+        self.currency = currency
+        self.description = description or ''
+        self.destination = f'ba_{random_id(24)}'
+        self.metadata = metadata or {}
+        self.method = method or 'standard'
+        self.source_type = source_type or 'card'
+        self.statement_descriptor = statement_descriptor or ''
+
+        two_days = 60 * 60 * 24 * 2
+        self.arrival_date = int(time.time() + two_days)
+
+        # As it was manually created as balance_transaction
+        # doesn't exist in localstripe yet, so we fake one
+        self.automatic = False
+        self.balance_transaction = f"txn_{random_id(24)}"
+
+        self.failure_balance_transaction = None
+        self.failure_code = None
+        self.failure_message = None
+        self.original_payout = None
+        self.reversed_by = None
+        self.status = 'pending'
+        self.type = 'bank_account'
+
+        schedule_webhook(Event('payout.created', self))
+
+        if status in ('paid', 'failed'):
+            self.status = status
+            if status == 'failed':
```

Comment:
If you follow my previous comment, here you can simplify the two `if` and keep only one.
Oh but now I see the `schedule_webhook(Event(f'payout.{status}', self))`.
I think `schedule_webhook(Event(f'payout.{status}', self))` would be better in his own dedicated `if`, splitter things like that help reading the code (even if variable are tested multiple time, that is not an issue).

Reply:
So you mean not to interpolate?

Reply:
What I had in mind is the following:
```py
if status in 'failed':
    self.failure_balance_transaction = ''	
    self.failure_code = 'could_not_process'	
    self.failure_message = \	         
        'The bank could not process this payout.'	    

if status in ('paid', 'failed'):
    schedule_webhook(Event(f'payout.{status}', self))
```
You see it's simpler because the logic of settings the "failure" fields is differentiated from the logic of sending webhook.
You can see them as two different features.
They share a common condition that activates them sure, but the most important is that they are two different features.
When the code shows it clearly, it becomes easier to read it (at least I think).